pkg/cloudprovider/aws/packing/packing.go

```diff
@@ -77,34 +78,46 @@ func (p *podPacker) Pack(ctx context.Context, pods []*v1.Pod) ([]*Packing, error
 }
 
 // TODO filter instance types based on node contraints like availability zones etc.
-func (p *podPacker) getNodeCapacities() []*nodeCapacity {
-	return nodeCapacities
+func (p *podPacker) getNodeCapacities(constraints *cloudprovider.Constraints) []*nodeCapacity {
+	result := make([]*nodeCapacity, 0)
+	for _, nc := range nodeCapacities {
+		ncc := nc.Copy()
+		kubeletOverhead := binpacking.CalculateKubeletOverhead(ncc.total)
+		if ok := ncc.reserve(resources.Merge(constraints.Overhead, kubeletOverhead)); !ok {
+			zap.S().Errorf("Failed to reserve kubelet overhead for node capacity type %v", ncc.instanceType)
+		}
+		result = append(result, ncc)
+	}
+	return result
+}
+
+type packingResult struct {
+	packed   []*v1.Pod
+	unpacked []*v1.Pod
 }
 
 // packWithLargestPod will try to pack max number of pods with largest pod in
 // pods across all available node capacities. It returns Packing: max pod count
 // that fit; with their node capacities and list of leftover pods
-func (p *podPacker) packWithLargestPod(pods []*v1.Pod) (*Packing, []*v1.Pod) {
+func (p *podPacker) packWithLargestPod(unpackedPods []*v1.Pod, constraints *cloudprovider.Constraints) (*Packing, []*v1.Pod) {
 	bestPackedPods := []*v1.Pod{}
 	bestCapacities := []*nodeCapacity{}
-	remainingPods := pods
-	// TODO reserve (Kubelet+ daemon sets) overhead for instance types
-	// TODO count number of pods created on an instance type
-	for _, nc := range p.getNodeCapacities() {
+	remainingPods := unpackedPods
+	for _, nc := range p.getNodeCapacities(constraints) {
 		// check how many pods we can fit with the available capacity
-		packedPods, unpackedPods := p.packPodsForCapacity(nc, pods)
-		if len(packedPods) == 0 {
+		result := p.packPodsForCapacity(nc, unpackedPods)
```

Comment:
did you mean to pass remaining pods here?

Reply:
Here we are passing all the pods which are unpacked and comparing, so, unpackedPods is fine.